File: src/main/java/ludwig/model/ReferenceNode.java
package ludwig.model;


public class ReferenceNode extends Node<ReferenceNode> {
    private final Node<?> ref;

    public ReferenceNode(Node<?> ref) {
        this.ref = ref;
    }

    public Node<?> ref() {
        return ref;
    }

    @Override
    public <T> T accept(NodeVisitor<T> visitor) {
        return visitor.visitReference(this);
    }

    @Override
    public String toString() {
        return ref.toString();
    }
}

File: src/main/java/ludwig/model/ReferenceNode.kt
package ludwig.model


class ReferenceNode(private val ref: Node<*>) : Node<ReferenceNode>() {

    fun ref(): Node<*> {
        return ref
    }

    override fun <T> accept(visitor: NodeVisitor<T>): T {
        return visitor.visitReference(this)
    }

    override fun toString(): String {
        return ref.toString()
    }
}
